### PR TITLE
[REF] Cleanup usage of CRM_Core_BAO_PrevNextCache::setItem and deprec…

### PR DESCRIPTION
**Pull request by seamuslee001**
…ate passing in array of values

Overview
----------------------------------------
This is a minor cleanup of the :;setItem function and specifically enforcing specific function params for each item to be added rather than an array of array of values that are just imploded

Before
----------------------------------------
_The current status. Please provide screenshots or gifs ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)) where appropriate._

After
----------------------------------------
_What has been changed. Please provide screenshots or gifs ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)) where appropriate._

Technical Details
----------------------------------------
_If the PR introduces noteworthy technical changes, please describe them here. Provide code snippets if necessary_

Comments
----------------------------------------
_Anything else you would like the reviewer to note_

### CODE REVIEW

File: CRM/Core/BAO/PrevNextCache.php [hunk 1]
```diff
@@ -315,14 +315,41 @@ public static function is_serialized($string) {
...
+  public static function setItem($values = [], $entity_table = NULL, $entity_id1 = NULL, $entity_id2 = NULL, $cacheKey = NULL, $data = NULL) {
+    if (!empty($values)) {
+      Civi::log()->warning('Deprecated code path. Values should not be set this is going away in the future in favour of specific function params for each column.', array('civi.tag' => 'deprecated'));
+      foreach ($values as $value) {
+        $valueArray = self::convertSetItemValues($value);
+        self::setItem([], $valueArray[0], $valueArray[1], $valueArray[2], $valueArray[3], $valueArray[4]);
+      }
+    }
+    else {
+      CRM_COre_DAO::executeQuery("INSERT INTO civicrm_prevnext_cache (entity_table, entity_id1, entity_id2, cacheKey, data) VALUES
```
Comment (pfigel) [inline CRM/Core/BAO/PrevNextCache.php]: `s/CRM_COre/CRM_Core/`

**Comment by pfigel** [comment]
@seamuslee001 I'm wondering if the `$values` and `$entity_table` parameters should be merged into one (e.g. `$entity_table_or_values`) to avoid having to pass an empty first param. We can then go with something like `if (is_array($entity_table_or_values)) { // old-style call } else { // new-style call }`. That, or use a new method ...

**Comment by seamuslee001** [comment]
@pfigel @eileenmcnaughton ok i have updated it as per your suggestion Patrick and also tried to put some more stringent nature into the setItem query

**Comment by eileenmcnaughton** [comment]
OK this looks OK - the test fail is unrelated & we do have solid test cover that passes through the code that calls the PrevNext::setItems

**Review by eileenmcnaughton — COMMENTED** [review]
Makes sense

**Pull request merged**